cluster-autoscaler/core/static_autoscaler.go

```diff
@@ -313,12 +313,14 @@ func (a *StaticAutoscaler) RunOnce(currentTime time.Time) errors.AutoscalerError
 		return errors.ToAutoscalerError(errors.CloudProviderError, err)
 	}
 
-	// Update node groups min/max after cloud provider refresh
+	// Update node groups min/max and maximum number of nodes being set for all node groups after cloud provider refresh
+	maxNodesCount := 0
 	for _, nodeGroup := range a.AutoscalingContext.CloudProvider.NodeGroups() {
 		metrics.UpdateNodeGroupMin(nodeGroup.Id(), nodeGroup.MinSize())
 		metrics.UpdateNodeGroupMax(nodeGroup.Id(), nodeGroup.MaxSize())
+		maxNodesCount += nodeGroup.MaxSize()
 	}
-
+	metrics.UpdateMaxNodesCount(maxNodesCount)
```

Comment:
Shouldn't it be `min(autoscalingOptions.MaxNodesTotal, maxNodesCount)` when `autoscalingOptions.MaxNodesTotal > 0`?

Reply:
`maxNodesTotal               = flag.Int("max-nodes-total", 0, "Maximum number of nodes in all node groups. Cluster autoscaler will not grow the cluster beyond this number.")`

I think they are equal. but because `maxNodesCount` is an env someone can set it lower than the actual number for any reason.
So `min` of them is better. I'll fix it.